### PR TITLE
nmstate: nmstate added unit tests

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/nmstate/nmstate_test.go [hunk 1]
```diff
@@ -1 +1,313 @@
...
+			assert.Equal(t, testNmState.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestNMStateNewBuilder(t *testing.T) {
+	testCases := []struct {
+		name              string
+		expectedErrorText string
+		client            bool
```
Comment (klaskosk) [inline pkg/nmstate/nmstate_test.go:109]: It doesn't look like the client field is ever used in the test. testSettings is always non-nil on line 126

File: pkg/nmstate/nmstate.go [hunk 2]
```diff
@@ -137,7 +157,9 @@ func (builder *Builder) Update(force bool) (*Builder, error) {
 
 	err := builder.apiClient.Update(context.TODO(), builder.Definition)
 
-	if err != nil {
+	if err == nil {
+		builder.Object = builder.Definition
+	} else if err != nil {
```
Comment (trewest) [inline pkg/nmstate/nmstate.go:162]: Why else if?

Comment (kononovn) [reply pkg/nmstate/nmstate.go:162]: +1. fixed

File: pkg/nmstate/nmstate_test.go [hunk 3]
```diff
@@ -1 +1,310 @@
...
+			name:              "",
+			expectedErrorText: "NMState 'name' cannot be empty",
+		},
+		{
+			name: "test1",
+		},
+	}
+	for _, testCase := range testCases {
+
+		testSettings := clients.GetTestClients(clients.TestClientParams{})
```
Comment (klaskosk) [inline pkg/nmstate/nmstate_test.go:123]: I think we should still test for when apiClient is nil, it just wasn't being tested before

**Review by klaskosk — APPROVED** [review]
/lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**